code/game/objects/effects/blessing.dm

```diff
@@ -6,20 +6,20 @@
 	anchored = TRUE
 	density = FALSE
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	var/image/holy_effect
 
 /obj/effect/blessing/Initialize(mapload)
 	. = ..()
-	for(var/obj/effect/blessing/B in loc)
-		if(B != src)
-			return INITIALIZE_HINT_QDEL
-		var/image/I = image(icon = 'icons/effects/effects.dmi', icon_state = "blessed", layer = ABOVE_OPEN_TURF_LAYER, loc = src)
-		I.alpha = 64
-		I.appearance_flags = RESET_ALPHA
-		add_alt_appearance(/datum/atom_hud/alternate_appearance/basic/blessedAware, "blessing", I)
+	holy_effect = image(icon = 'icons/effects/effects.dmi', icon_state = "blessed", layer = ABOVE_OPEN_TURF_LAYER, loc = src)
+	holy_effect.alpha = 64
+	holy_effect.appearance_flags = RESET_ALPHA
+	GLOB.cimg_controller.stack_client_images(CIMG_KEY_HOLYTURF, holy_effect)
 	RegisterSignal(loc, COMSIG_ATOM_INTERCEPT_TELEPORT, PROC_REF(block_cult_teleport))
 
 /obj/effect/blessing/Destroy()
 	UnregisterSignal(loc, COMSIG_ATOM_INTERCEPT_TELEPORT)
+	GLOB.cimg_controller.cut_client_images(CIMG_KEY_HOLYTURF, holy_effect)
+	holy_effect = null
```

Comment:
I think this highlights one major disadvantage with this system being that you are now responsible for handling the cleaning up of alt appearances yourself. If an atom is deleted, then any overriding images attached to it should be able to just be cleaned up by itself. Whether thats handled through a lazy list, or has an "alt appearance" component wrapper which would be nice

Reply:
See the overall review comment, components/elements might not work if these are intended to be changed a lot (accessing a component/element through GetComponent or storing the component should be banned)